read out portage group from Portage itself, not to get false root checks

http://bugs.gentoo.org/show_bug.cgi?id=278478

revised version by Tim Spriggs, bug #237811

--- autounmask
+++ autounmask
@@ -83,8 +83,9 @@
 if ($< > 0) {
 	my @userGroups = split(' ',`groups`);
 	my $userPermCheck = 0;
+	chomp(my $portageGroup = `python -c 'from portage.const_autotool import portagegroup; print portagegroup,'`);
 	foreach (@userGroups) {
-		if ($_ eq 'portage') {
+		if ($_ eq $portageGroup) {
 			$userPermCheck = 1;
 			last;
 		}
